app/src/main/java/com/hedvig/app/feature/swedishbankid/sign/usecase/ManuallyRecheckSwedishBankIdSignStatusUseCase.kt

```diff
@@ -0,0 +1,26 @@
+package com.hedvig.app.feature.swedishbankid.sign.usecase
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.fragment.SignStatusFragment
+import com.hedvig.android.owldroid.graphql.SignStatusQuery
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+import e
+
+class ManuallyRecheckSwedishBankIdSignStatusUseCase(
+    private val apolloClient: ApolloClient,
+) {
+    suspend operator fun invoke(): SignStatusFragment? {
+        val result = apolloClient
+            .query(SignStatusQuery())
+            .safeQuery()
+
+        return when (result) {
+            is QueryResult.Error -> {
+                e { "Error manually rechecking sign status: ${result.message}" }
```

Comment:
Perhaps this could return a result as well, containing `Success` with `status.collectStatus?.status`, `status.collectStatus?.code` and `status.signState` (which we use in `SwedishBankIdSignViewModel.toViewStateOrNull`. And in addition to Success - Error and Empty states?

Reply:
Given that this only re-fetches the data in case of missed subscription messages, it is slightly more lenient on purpose.